EIPS/eip-7756.md

```diff
@@ -0,0 +1,175 @@
+- The `memory` or `storage` members may be omitted if they are empty or the client does not produce them.
+- The `memSize` member MUST be present regardless of `memory` support. Even when `memSize` is zero.
+- The `error` and `returnData` members can be omitted if they are empty.
+- If the prior trace line had an exceptional halt, `error` should identify the halt.  Otherwise, if a `REVERT` operation occurred it should contain the revert reason.
+- The `storage` member should only include items read or written via `SSTORE` or `SLOAD`, and not the account's entire memory.
+- Clients SHOULD implement a way to disable recording the storage as the stateroot includes all storage updates.
+- Clients SHOULD output the members in the same order as listed in this EIP.
+
+*Example:*
+
+```
+{"pc":0,"op":96,"gas":"0x2540be400","gasCost":"0x3","memory":"0x","memSize":0,"stack":[],"depth":1,"error":null,"opName":"PUSH1"}
+```
+
+### Summary Line
+
+At the end of execution, the client should print summary info. This summary MUST as a single JSON object.
+
+This info SHOULD have the following members.
+
+| Name        | Type       | Required | Explanation                                            |
+|-------------|------------|----------|--------------------------------------------------------|
+| `stateRoot` | Hex-String | Yes      | Root of the state trie after executing the transaction |
+| `output`    | Hex-String | Yes      | Return values of the function                          |
+| `gasUsed`   | Hex-Number | Yes      | All gas used by the transaction                        |
+| `pass`      | Boolean    | Yes      | If the tx was successful, or if the test passed        |
+| `time`      | Number     | No       | Time in nanoseconds needed to execute the transaction  |
+| `fork`      | String     | No       | Name of the fork rules used for execution              |
+
+*Example*:
+
+```
+{"stateRoot":"0xd4c577737f5d20207d338c360c42d3af78de54812720e3339f7b27293ef195b7","output":"","gasUsed":"0x3","pass":"true","time":141485}
+```
+
+## Rationale
+
+This EIP is an extension of the EIP-3155 tracing features that has been in use for years. Rather than dramatically re-boot the feature, the information was added to the existing traces.
+
+A "mini" trace was contemplated to allow for tracing to be included in tools such as `t8n` and to allow for more efficient RPC tracing calls, but that seemed sufficiently different that it would be a stand-alone EIP rather than an EIP that adds features to the existing tracing capabilities.
+
+## Backwards Compatibility
+
+Clients emitting tracing JSON for uncontained "legacy" contracts will produce a compatible trace, except as outlined below
+
+### Changes from EIP-3155
+
+- The term Client Under Test or CUT has been replaced simply with "client."
+
+### Additions to EIP-3155
+
+- The `immediate` member was added to support the large number of instructions that contain immediate operations.
```

Comment:
What should `immediate` produce for the RJUMPV? Should it report the entire table?

Reply:
Hmm... up above I added a subpoint allowing just the table length as it can be implied from the trace.